applications/launchpad/backend/src/docker/mod.rs

```diff
@@ -168,3 +172,45 @@ pub async fn shutdown_all_containers(workspace_name: String, docker: &Docker) ->
     }
     Ok(())
 }
+
+pub async fn listen_progress_info() -> Result<(), DockerWrapperError> {
+    info!("Listen for base node info....");
+    let mut client = GrpcBaseNodeClient::new();
+    loop {
+        let connected = client.connected().await;
+        if connected {
+            break;
+        } else {
+            continue;
+        }
```

Comment:
I still dislike this spinlock style of getting a connection.
If you want a `get_connection_or_wait_until_death_heat_of_the_universe` type functionality, it should at least be in its own method so we can add more advanced logic like exponential backoff, etc.